core/common/src/main/java/alluxio/metrics/MetricsSystem.java

```diff
@@ -186,6 +198,21 @@ public static String getMetricNameWithUniqueId(String instance, String name) {
         .join(instance, NetworkAddressUtils.getLocalHostName().replace('.', '_'), name);
   }
 
+  /**
+   * Builds unique metric registry names with unique ID (set to host name) and user. The pattern is
+   * instance.hostname.metricName.username.
+   *
+   * @param instance the instance name
+   * @param userName the user name
+   * @param name the metric name
+   * @return the metric registry name
+   */
+  public static String getMetricNameWithUniqueIdAndUser(String instance, String userName,
+      String name) {
+    return Joiner.on(".").join(instance, NetworkAddressUtils.getLocalHostName().replace('.', '_'),
```

Comment:
can we include the tag names, e.g. "instance.hostname.metricName;user=username"? Otherwise backwards compatibility and metric name readability will become problems if we add additional tags.